### DISCUSSION TITLE
SetLoadersPath configuration issue

### DISCUSSION THREAD

**Discussion by aiden1536** [K-Society/KSociety.SharpCubeProgrammer#143] [Q&A] [answered]
When I initially didn't use SetLoadersPath, an error message "Error: Unable to list supported devices" appeared after executing ConnectStLink. Therefore, I needed to set SetLoadersPath("./."). 

However, after setting it, the program crashed when executing ConnectStLink (System.AccessViolationException: 'Attempted to read or write protected memory. This is often an indication that other memory is corrupt.'). 

Did I set it incorrectly? Has anyone encountered the same issue and can help me resolve it? Thank you.

**Comment by maniglia** [comment]
Hello @aiden1536 
by default it is not necessary to use the SetLoadersPath function.

Are you using the latest version of the library?

Please use PackageReference.

**Reply by aiden1536** [reply]
I am using the latest version, but when I don't use SetLoadersPath, it prompts "Error: Unable to list supported devices". So I guessed that I need to specify the path to the "ExternalLoader" and "FlashLoader" folders.

**Reply by maniglia** [reply]
Please can you share a project that reproduces this problem?

**Reply by aiden1536** [reply]
Currently, I created a new test project, and using the default path works fine. However, the previous project I'm developing is an older .NET Framework 4.8 project. I am unable to reproduce the same issue in the new project, but the problem with the default path not working seems to be due to the absence of the ST folder in the built path after the build, which leads to the "Error: Unable to list supported devices" error. I have manually copied the folder to resolve this issue.

However, I still need to mention that my problem occurs when I use the method "SetLoadersPath" and then use "ConnectStLink," which causes a crash. Could it be an issue with the path encoding?

**Reply by maniglia** [reply]
Thanks for this information.
I'll try to do some tests with the .NET Framework 4.8 as soon as I find the time to do so.

**Reply by maniglia** [reply]
Have you already tried deleting the contents of the _obj_ folder and then recompiling the project?

**Reply by maniglia** [reply]
I did some tests but I can't reproduce your problem.

**Reply by aiden1536** [reply]
I tried cleaning the project and rebuilding it, but the project still cannot automatically generate the ST folder, leading to the inability to recognize devices when reading. Please understand this issue. However, when I create a new .NET 6 project, the ST folder appears. I'm not sure what the problem is.

**Reply by maniglia** [reply]
Can you share your project?
I have no elements to work on.

**Reply by aiden1536** [reply]
Let's test it and see if it works.

[StCubeDemo.zip](https://github.com/user-attachments/files/15881149/StCubeDemo.zip)

**Reply by maniglia** [reply, marked as answer]
PackageReference must be used.
[StCubeDemo.zip](https://github.com/user-attachments/files/15892325/StCubeDemo.zip)
[PackageReference](https://learn.microsoft.com/en-us/nuget/consume-packages/migrate-packages-config-to-package-reference)

**Reply by aiden1536** [reply]
So there's such a difference, thank you for your support.
